### PR TITLE
PR - Extract steam scrape

### PR DESCRIPTION
**Pull request by Leon-JS02**
Added functions to use an additional URL to extract:
- Game genres
- Game tags
- Game description
- Supported operating systems

Added tests.

### CODE REVIEW

File: web_scraping/steam_scraper/scrape_steam.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import bs4
 
 STEAM_NEW_RELEASE_URL = "https://store.steampowered.com/search/?sort_by=Released_DESC&category1=998"
+STEAM_APP_URL = "https://store.steampowered.com/app/"
 TIMEOUT = 10
```
Comment (mehrdadhalali) [inline web_scraping/steam_scraper/scrape_steam.py:10]: Good thinking! Adding a timeout limit is important.

File: web_scraping/steam_scraper/scrape_steam.py [hunk 2]
```diff
@@ -67,20 +68,85 @@ def parse_price(game_listing: bs4.Tag) -> str:
...
+    if description_tag:
+        return description_tag.text.strip()
+    return "No description available."
+
+
+def scrape_game_tags(app_soup: bs4.BeautifulSoup) -> list[str]:
+    """Scrapes a game's Steam app page for tags."""
+    tag_elements = app_soup.find_all('a', class_='app_tag')
+    tags = [tag.text.strip() for tag in tag_elements]
+    return tags if tags else ["No tags available."]
```
Comment (mehrdadhalali) [inline web_scraping/steam_scraper/scrape_steam.py:88]: Nice backup plan if there are no tags!

File: web_scraping/steam_scraper/scrape_steam.py [hunk 3]
```diff
@@ -67,20 +68,85 @@ def parse_price(game_listing: bs4.Tag) -> str:
...
+                  for genre in genre_elements if "genre" in genre['href']]
+    return genres if genres else []
+
+
+def scrape_game_operating_systems(app_soup: bs4.BeautifulSoup) -> list[str]:
+    """Scrapes a game's Steam app page for supported operating systems."""
+    platform_mapping = {
+        'win': 'Windows',
+        'mac': 'Mac',
+        'linux': 'Linux'
```
Comment (mehrdadhalali) [inline web_scraping/steam_scraper/scrape_steam.py:106]: Could go even further and only look at the first character lol

**Review by mehrdadhalali — APPROVED** [review]

File: web_scraping/steam_scraper/scrape_steam.py (hunk 3 above)
Comment (Leon-JS02) [reply web_scraping/steam_scraper/scrape_steam.py:106]: Good idea. I believe dictionaries are constant lookup time regardless, though. 💪

File: web_scraping/steam_scraper/scrape_steam.py [hunk 4]
```diff
@@ -67,20 +68,85 @@ def parse_price(game_listing: bs4.Tag) -> str:
...
+    os = scrape_game_operating_systems(app_soup)
+    genres = scrape_game_genres(app_soup)
+    tags = scrape_game_tags(app_soup)
+
     return {
-        'app_id': parse_app_id(game_listing),
-        'url': parse_game_url(game_listing),
-        'img_url': parse_image_url(game_listing),
         'title': parse_title(game_listing),
+        'description': description,
```
Comment (MichaelHassan) [inline web_scraping/steam_scraper/scrape_steam.py:134]: Could this be shortened by  'description': scrape_game_description(app_soup) . Not overly important, just a suggestion for aesthetics.

**Review by MichaelHassan — APPROVED** [review]
Good broken up functions. Maybe some stylistic changes could be made, however overall happy to approve

**Pull request merged**